src/components/authorise/authorise-get-controller.ts

```diff
@@ -0,0 +1,74 @@
+import { Request, Response } from "express";
+import { logger } from "../../logger";
+import { parseAuthQueryParams } from "../../parse/parse-auth-request-query-params";
+import { AuthoriseRequestError } from "../../errors/authorise-request-error";
+import { BadRequestError } from "../../errors/bad-request-error";
+import { ParseRequestError } from "../../errors/parse-request-error";
+import { Config } from "../../config";
+import { MissingParameterError } from "../../errors/missing-parameter-error";
+import { DuplicateParameterError } from "../../errors/duplicate-parameter-error";
+
+export const authoriseGetController = (req: Request, res: Response): void => {
+  const config = Config.getInstance();
+
+  try {
+    const parsedAuthRequestParams = parseAuthQueryParams(req.query);
+
+    const authCode = generateAuthCode();
+
+    config.addToAuthCodeRequestParamsStore(authCode, {
+      claims: parsedAuthRequestParams.claims,
+      nonce: parsedAuthRequestParams.nonce,
+      redirectUri: parsedAuthRequestParams.redirect_uri,
+      scopes: parsedAuthRequestParams.scope,
+    });
+
+    res.redirect(
+      `${parsedAuthRequestParams.redirect_uri}?code=${authCode}&state=${parsedAuthRequestParams.state}&nonce=${parsedAuthRequestParams.nonce}`
```

Comment:
I'm pretty sure we don't return the nonce in this request?

Reply:
Yeah we don't, I have no idea why I added this 🤦

Reply:
Fixed